src/realm/array.cpp

```diff
@@ -1971,6 +1971,106 @@ std::pair<ref_type, size_t> Array::get_to_dot_parent(size_t ndx_in_parent) const
     return std::make_pair(get_ref(), ndx_in_parent);
 }
 
+
```

Comment:
It would have been nice if you would not have moved the code so much around. You could just have moved an `#endif`